js/apps/system/_admin/aardvark/APP/frontend/js/views/documentView.js

```diff
@@ -540,7 +534,7 @@
       $('#subNavigationBar .breadcrumb').html(
         '<a href="#collection/' + name[1] + '/documents/1">Collection: ' + (name[1].length > 64 ? name[1].substr(0, 64) + "..." : name[1]) + '</a>' +
         '<i class="fa fa-chevron-right"></i>' +
-        this.type.charAt(0).toUpperCase() + this.type.slice(1) + ': ' + name[2]
+        this.type.charAt(0).toUpperCase() + this.type.slice(1) + ': ' + decodeURI(name[2])
```

Comment:
Isn't `decodeURIComponent` the correct call to make here? Plus, must the result be encoded regarding HTML entities?

Reply:
`encodeURIComponent()` is said to encode 11 characters more than `encodeURI()`: `# $ & + , / : ; = ? @`.

Allowed characters in fragment identifier are apparently: `! $ & ' ( ) * + , ; = - . _ ~ : @ / ?` (as well as alpha, digit and `%` with two hex digits).

`#` seems like the only character that would require encoding from `encodeURIComponent()`, but we disallow it in document keys. So both encoding schemes should actually be fine, but they should obviously be used consistently.

In HTML attribute values, `'` needs to be encoded as `&#39;` or `&apos;` and `"` as `&#34;` or `&quot;`.
`&`, `<` and `>` need to be encoded as `&amp;`, `&lt;` and `&gt;` everywhere.

- https://stackoverflow.com/questions/75980/when-are-you-supposed-to-use-escape-instead-of-encodeuri-encodeuricomponent
- https://stackoverflow.com/questions/2849756/list-of-valid-characters-for-the-fragment-identifier-in-an-url
- https://tools.ietf.org/html/rfc3986#appendix-A
- https://stackoverflow.com/questions/7381974/which-characters-need-to-be-escaped-in-html